implants/lib/eldritch/src/sys/exec_impl.rs

```diff
@@ -183,6 +227,30 @@ mod tests {
         }
         Ok(())
     }
+
+    #[test]
+    fn test_sys_exec_disown_no_defunct() -> anyhow::Result<()> {
+        init_logging();
+
+        if cfg!(target_os = "linux")
+            || cfg!(target_os = "ios")
+            || cfg!(target_os = "macos")
+            || cfg!(target_os = "android")
+            || cfg!(target_os = "freebsd")
+            || cfg!(target_os = "openbsd")
+            || cfg!(target_os = "netbsd")
+        {
+            let _res = handle_exec(
+                String::from("/bin/sleep"),
+                vec!["3".to_string()],
+                Some(true),
+            )?;
+            let res = get_zombie_child_processes(process::id())?;
```

Comment:
should you wait 3 seconds here?

Reply:
The issue we were seeing was that the intermediate process was hanging in a defunct state not the grandchild.